### PR TITLE
exp: dorq task artifacts

### PR DESCRIPTION
**Pull request by alexjuda**
# The problem

* We need a command for accessing intermediate workflow values.

# This PR's solution

* New command: `dorq task results`
* Changed `--download-dir` behavior for `dorq wf results --download-dir <dir>` to make it consistent with the new command. Now, the artifacts will be stored under `<dir>/wf_results/<wf_run_id><ext>` and `<dir>/task_results/<wf_run_id>/<task_invocation_id>/<output_number><ext>`, respectively.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: src/orquestra/sdk/_base/cli/_dorq/_task/_logs.py [hunk 1]
```diff
@@ -46,29 +46,16 @@ def __init__(
...
 
-    def on_cmd_call(
-        self,
-        wf_run_id: t.Optional[WorkflowRunId],
-        task_invocation_id: t.Optional[TaskInvocationId],
-        fn_name: t.Optional[str],
-        config: t.Optional[ConfigName],
-        download_dir: t.Optional[Path],
-    ):
+    def on_cmd_call(self, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/cli/_dorq/_task/_logs.py:49]: to be clear: I like the change, I'm just curious as to what precipitated it :)

[the remark this answers is not in the document]
Comment (alexjuda) [reply src/orquestra/sdk/_base/cli/_dorq/_task/_logs.py:49]: I've noticed we have 3 separate places where we have the exact same parameter list, repeated for every command:
1. in `click` command functions in `_entry.py`.
2. in `_foo.Action.on_cmd_call()`.
3. in `_foo.Action._on_cmd_call_with_exceptions()`.

Additional considerations:
* all three places are directly related and should be in perfect sync
* there are no external users of these functions/methods
* static code analyzers seem to resolve the param list on hover anyway
* It's good to strive for [SSOT](https://en.wikipedia.org//wiki/Single_source_of_truth). I've found myself editing one parameter list but forgetting about the others

I'm not a fan of varargs in general, but combining the all above I believe this is a good use case for `*args/**kwargs` :)

**Comment by alexjuda** [comment]
Changes after the rebase: https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/26/files/2f8763a00c7ccc16b24b0d3bbfea996a673a23ac..47efa95eb914a5fe3075265f2a58e27e786a8b7c

File: src/orquestra/sdk/_base/cli/_dorq/_task/_logs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply src/orquestra/sdk/_base/cli/_dorq/_task/_logs.py:49]: Ah that makes sense. Thanks for explaining!

**Review by BenjaminMummery — APPROVED** [review]
LGTM :)

**Pull request merged**